e2e_test/streaming/bug_fixes/issue_11915.slt

```diff
@@ -0,0 +1,41 @@
+# https://github.com/risingwavelabs/risingwave/issues/11915
+
+statement ok
+create table t(x int);
+
+statement ok
+create materialized view mv as select x, generate_series(1, 2, x) from t;
+
+# x = 0 causes generate_series(1, 2, x) to return an error.
+statement ok
+insert into t values (0), (1);
+
+statement ok
+flush;
+
+# Output 0 row when the set-returning function returns error.
```

Comment:
There was an discussion on the behavior in this case. https://github.com/risingwavelabs/risingwave/issues/12474#issuecomment-1763368554 cc @fuyufjh 

Not sure which one we should adopt.

Reply:
The default behavior for a table function not being called (e.g. with null input) is returning no rows. So I vote for this for the sake of consistency.

Reply:
> The default behavior for a table function not being called (e.g. with null input) is returning no rows. So I vote for this for the sake of consistency.

I agree that these 2 cases should be consistent

Reply:
This is Option 1 in the original issue. And I agree it is consistent with null input. In other words, the table function returns `null` (which is similar to `{}` but not `{null}`) on error.

Related edges cases on the differences among `null`, `{}` and `{null}`:
* `array_agg` function on zero input rows return no rows with `group by`, but a `null` row with scalar agg. #15590
* Scalar subquery returning zero rows result in a `null` row. Array constructing subquery returning zero rows result in a `{}`. #15593

^ Just mentioning these cases. Not suggesting any value is always preferred than the other 2.